src/main/java/org/terasology/launcher/game/GameService.java

```diff
@@ -0,0 +1,164 @@
+ * the game failed to start or it terminates with an error. Whether it succeeded or failed, the service will then
+ * reset to {@link State#READY READY} to be started again.
+ * <p>
+ * This service {@linkplain #cancel() does not support cancellation}.
+ * <ul>
+ *   <li>For details on how the arguments to the process are constructed, see the source for {@link GameStarter}.
+ *   <li>For details on how output from the game process is treated, see {@link RunGameTask}.
+ * </ul>
+ */
+public class GameService extends Service<Boolean> {
+    private static final Logger logger = LoggerFactory.getLogger(GameService.class);
+
+    private Path gamePath;
+    private BaseLauncherSettings settings;
+
+    public GameService() {
+        setExecutor(Executors.newSingleThreadExecutor(
+            new ThreadFactoryBuilder()
+                    .setNameFormat("GameService-%d")
+                    .setDaemon(true)
+                    .setUncaughtExceptionHandler(this::exceptionHandler)
+                    .build()
+        ));
+    }
+
+    /**
+     * Start a new game process with these settings.
+     *
+     * @param gamePath the directory under which we will find libs/Teresology.jar, also used as the process's
+     *     working directory
+     * @param settings supplies other settings relevant to configuring a process
+     */
+    @SuppressWarnings("checkstyle:HiddenField")
+    public void start(Path gamePath, BaseLauncherSettings settings) {
+        this.gamePath = gamePath;
+        this.settings = settings;
+
+        start();
+    }
+
+    /**
+     * Use {@link #start(Path, BaseLauncherSettings)} instead.
+     * <p>
+     * It is an error to call this method before providing the configuration.
+     */
+    @Override
+    public void start() {
+        super.start();
+    }
+
+    /**
+     * Cancellation is unsupported. Do not attempt this method.
+     * <p>
+     * Rationale: We do not terminate a running game process, and we don't want to lose our thread keeping track of
+     * the process while it's still live. If we did, our “is a game already running?” logic would fail.
+     * <p>
+     * If you are using this with some kind of generic Service Manager that always invokes this method as part of an
+     * orderly shutdown, that could be a good reason to change this from throwing an exception to logging a warning.
+     * Until then, this fails loudly so you won't call this method thinking it does something and then become confused
+     * when nothing happens.
+     *
+     * @throws UnsupportedOperationException always
+     */
+    @Override
+    public boolean cancel() {
+        throw new UnsupportedOperationException("GameService does not cancel.");
+    }
+
+    /**
+     * Restarting is unsupported. Do not attempt this method.
+     * <p>
+     * See {@link #cancel()} for rationale.
+     */
+    @Override
+    public void restart() {
+        super.restart();
```

Comment:
If it is unsupported, why not 
```java
throw new UnsupportedOperationException("GameService cannot be restarted.");
```

Reply:
The only reason `restart` is unsupported is because `cancel` is unsupported, and `cancel` will throw that.

Which leaves us much the same [question](https://github.com/MovingBlocks/TerasologyLauncher/pull/571/files#r432520199) as on the zero-argument form of `start()`: Is identifying that condition a few stack frames earlier worth a few lines of redundant code?